prog/weaver/http.go

```diff
@@ -19,6 +19,8 @@ import (
 	weave "github.com/weaveworks/weave/router"
 )
 
+var allConnectionStates = []string{"established", "pending", "retrying", "failed", "connecting"}
```

Comment:
Nitpicking, but I'd prefer `mesh` to provide the list of all possible connection states. Also, this list does not include the `waiting` state (although the if-branch which creates it is marked with the `"shouldn't happen"` comment).